pkg/katapult/virtual_machine_group.go

```diff
@@ -14,13 +15,53 @@ type VirtualMachineGroup struct {
 	CreatedAt *timestamp.Timestamp `json:"created_at,omitempty"`
 }
 
-type VirtualMachineGroupsClient struct {
-	client   *apiClient
-	basePath *url.URL
+func (s *VirtualMachineGroup) lookupReference() *VirtualMachineGroup {
+	if s == nil {
+		return nil
+	}
+
+	return &VirtualMachineGroup{ID: s.ID}
+}
+
+func (s *VirtualMachineGroup) queryValues() *url.Values {
+	v := &url.Values{}
+
+	if s != nil {
+		v.Set("virtual_machine_group[id]", s.ID)
+	}
+
+	return v
+}
+
+type VirtualMachineGroupCreateArguments struct {
+	Name      string
+	Segregate *bool
+}
+
+type VirtualMachineGroupUpdateArguments struct {
+	Name      string
+	Segregate *bool
+}
+
+type virtualMachineGroupCreateRequest struct {
+	Organization *Organization `json:"organization,omitempty"`
+	Name         string        `json:"name,omitempty"`
+	Segregate    *bool         `json:"segregate,omitempty"`
+}
+
+type virtualMachineGroupUpdateRequest struct {
+	VirtualMachineGroup *VirtualMachineGroup `json:"virtual_machine_group,omitempty"`
+	Name                string               `json:"name,omitempty"`
+	Segregate           *bool                `json:"segregate,omitempty"`
 }
```

Comment:
These classes which have `json` field tags and are used to marshal/unmarshal requests should have each have JSON marshaling tests like the `VirtualMachineGroup` struct.

The marshaling test for the `ipAddressCreateRequest` struct should serve as a good reference, as it also has a boolean pointer type which needs an explicit tests case with a `false` value: https://github.com/krystal/go-katapult/blob/master/pkg/katapult/ip_address_test.go#L281-L310

These tests will not pass on their own though, as they use a common Go convention of "golden" files. So once the test is written, run `make test-update-golen` to execute the test suite in a way that golden files are written if they're missing. After that you can then run the tests like normal.

P.S. Sorry I missed this with the `virtualMachineGroupsResponseBody` in #76.